tests/testthat/test_nullModel.R

```diff
@@ -317,3 +359,76 @@ test_that("data.tables are supported", {
+    expect_equal("x ~ (1|r) + (1|s) + var(g)",
+                 .modelString(outcome, NULL, random, group_var))
+    expect_equal("x ~ a + b + var(g)",
+                 .modelString(outcome, covars, NULL, group_var))
+    expect_equal("x ~ ",
+                 .modelString(outcome, NULL, NULL, NULL))
+
+    # With inverse_normal = TRUE
+    expect_equal("rankInvNorm(resid(x)) ~ a + b + (1|r) + (1|s) + var(g)",
+                 .modelString(outcome, covars, random, group_var, inverse_normal = TRUE))
+    expect_equal("rankInvNorm(resid(x)) ~ a + (1|r) + var(g)",
+                 .modelString(outcome, covars[1], random[1], group_var, inverse_normal = TRUE))
+    expect_equal("rankInvNorm(resid(x)) ~ (1|r) + (1|s) + var(g)",
+                 .modelString(outcome, NULL, random, group_var, inverse_normal = TRUE))
+    expect_equal("rankInvNorm(resid(x)) ~ a + b + var(g)",
+                 .modelString(outcome, covars, NULL, group_var, inverse_normal = TRUE))
+    expect_equal("rankInvNorm(resid(x)) ~ ",
+                 .modelString(outcome, NULL, NULL, NULL, inverse_normal = TRUE))
+})
+
+test_that(".modelOutcomeString", {
+  outcome <- "x"
+  expect_equal(.modelOutcomeString(outcome), "x")
+  expect_equal(.modelOutcomeString(outcome, inverse_normal = TRUE),  "rankInvNorm(resid(x))")
+})
+
+
+
+#if(FALSE){
+test_that("update conditional model", {
+  set.seed(23); a <- rnorm(100)
+  set.seed(57); G <- matrix(rnorm(100, 100,1))
+  dat <- data.frame(sample.id=make.unique(rep(letters, length.out = 100), sep = ""),
+                    a=a,
+                    b=c(rep("a",50), rep("b", 50)),
+                    var_1=G[,1],
+                    stringsAsFactors=FALSE)
+  dat <- AnnotatedDataFrame(dat)
+  set.seed(24); covMat <- crossprod(matrix(rnorm(10000,sd=0.05),100,100))
+  dimnames(covMat) <- list(dat$sample.id, dat$sample.id)
+  nullmod <- fitNullModel(dat, outcome="a", covars="b", cov.mat=covMat, verbose=FALSE)
+
+  # With genotype.
+  nullmod_geno <- fitNullModel(dat, outcome="a", covars=c("b", "var_1"), cov.mat=covMat, verbose=FALSE)
+  # Updated null model with unnamed genotype.
+  nullmod_update <- updateNullModCond(nullmod, G, covMatList=covMat, verbose=FALSE)
+
+  expect_equivalent(nullmod_update$varComp, nullmod_geno$varComp, tolerance=1e-4)
+  expect_equivalent(nullmod_update$fixef, nullmod_geno$fixef, tolerance=1e-4)
+  expect_equivalent(nullmod_update$cholSigmaInv, nullmod_geno$cholSigmaInv, tolerance=1e-4)
+  expect_equivalent(nullmod_update$varCompCov, nullmod_geno$varCompCov, tolerance=1e-4)
+  expect_equal(nullmod_update$model, nullmod_geno$model)
+  expect_equal(nullmod_update$covars, nullmod_geno$covars)
+
+  # Genotype matrix with colnames.
+  colnames(G) <- "variant"
+  nullmod_update <- updateNullModCond(nullmod, G, covMatList=covMat, verbose=FALSE)
+  expect_equal(nullmod_update$model, "a ~ b + variant + (1|A)")
+  expect_equal(nullmod_update$covars, c("b", "variant"))
+
+})
+#}
```

Comment:
This test was not run due to the `if(FALSE)` statement, and it in a different test_* source file. I rewrote it a bit and moved it to the test_nullModel source file. Is there a reason to not run it?

Reply:
I think it was giving errors in R CMD check that I did not understand (maybe only on the Bioconductor build system), and since we don't use or export this function anyway, I just stopped running the test as a temporary fix. If it works after rewriting, it should be run.

Reply:
It runs in R CMD check - we can take it out again if it fails in the Bioconductor build system and we don't know how to fix it.